hydrolib/core/io/obscrosssection/models.py

```diff
@@ -0,0 +1,86 @@
+        fileversion: Optional[str] = Field(
+            "File version. Do not edit this.", alias="fileVersion"
+        )
+        filetype: Optional[str] = Field(
+            "File type. Should be 'obsCross'. Do not edit this.", alias="fileType"
+        )
+
+    comments: Comments = Comments()
+    fileversion: str = Field("2.00", alias="fileVersion")
+    filetype: Literal["obsCross"] = Field("obsCross", alias="fileType")
+
+
+class ObservationPointCrossSection(INIBasedModel):
+    """
+    The cross section of an observation point that is included in the
+    observation cross section file.
+
+    All lowercased attributes match with the observation point cross
+    section output as described in [UM Sec.F2.4.1]
+    (https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#subsubsection.F.2.4.1)
+    """
+
+    class Comments(INIBasedModel.Comments):
+        name: Optional[str] = "Name of the cross section (max. 255 characters)."
+        branchid: Optional[str] = Field(
+            "(optional) Branch on which the cross section is located.", alias="branchId"
+        )
+        chainage: Optional[str] = "(optional) Location on the branch (m)."
+        numcoordinates: Optional[str] = Field(
+            "(optional) Number of values in xCoordinates and yCoordinates. "
+            "This value should be greater than or equal to 2.",
+            alias="numCoordinates",
+        )
+        xcoordinates: Optional[str] = Field(
+            "(optional) x-coordinates of the cross section line. "
+            "(number of values = numCoordinates)",
+            alias="xCoordinates",
+        )
+        ycoordinates: Optional[str] = Field(
+            "(optional) y-coordinates of the cross section line. "
+            "(number of values = numCoordinates)",
+            alias="yCoordinates",
+        )
+
+    comments: Comments = Comments()
+    _header: Literal["ObservationCrossSection"] = "ObservationCrossSection"
+    name: str = Field(max_length=255, alias="name")
+    branchid: Optional[str] = Field(alias="branchId")
+    chainage: Optional[float] = Field(alias="chaiange")
+    numcoordinates: Optional[int] = Field(alias="numCoordinates")
+    xcoordinates: Optional[List[float]] = Field(alias="xCoordinates")
+    ycoordinates: Optional[List[float]] = Field(alias="yCoordinates")
+
+    _split_to_list = get_split_string_on_delimiter_validator(
+        "xcoordinates", "ycoordinates"
+    )
+
+    _location_validator = get_location_specification_rootvalidator(allow_nodeid=False)
+
+    _number_of_coordinates_validator = get_number_of_coordinates_validator()
+
+
+class ObservationPointCrossSectionModel(INIModel):
+    """
+    The overall observation point crosssection model that contains the contents
+    of one observation point crosssection file.
+    """
+
+    general: ObservationPointCrossSectionGeneral = ObservationPointCrossSectionGeneral()
+    crosssections: List[ObservationPointCrossSection] = []
```

Comment:
The agreed convention for these lists is to call the field the singular form of the object in the list, so in this case:

```suggestion
    observationpointcrosssection: List[ObservationPointCrossSection] = []
```

(but it should be `observationcrosssection`)

I don't remember why exactly, but I will ask @arthurvd or @BeardedPlatypus when they get back.
There was a specific reason for it.

Reply:
Fixed.